src/cascade/input_data/db/bundle.py

```diff
@@ -0,0 +1,238 @@
+         uncertainty_type_id ,
+         uncertainty_type_value ,
+         mean ,
+         lower ,
+         upper ,
+         standard_error ,
+         effective_sample_size ,
+         sample_size ,
+         cases ,
+         design_effect,
+         outlier_type_id
+        FROM
+         epi.bundle_dismod
+        WHERE
+         bundle_id = %(bundle_id)s and
+         input_type_id NOT IN(5,6) and
+         outlier_type_id IN (0,1)
+         """
+    with cursor(database=database) as c:
+        c.execute(query, args={"bundle_id": bundle_id})
+        bundle_rows = list(c)
+        CODELOG.debug(
+            f"Downloaded {len(bundle_rows)} lines of bundle_id {bundle_id} from '{database}'"
+        )
+
+    return bundle_rows
+
+
+def _get_tier_2_study_covariates(execution_context):
+    """Downloads the tier 2 study covariate mappings for the bundle associated with the current model_version_id.
+    """
+    database = execution_context.parameters.bundle_database
+
+    bundle_id = _get_bundle_id(execution_context)
+
+    query = f"""
+    SELECT
+        bundle_id,
+        seq,
+        study_covariate_id
+    FROM
+        epi.bundle_dismod_study_covariate
+    WHERE
+        bundle_id = %(bundle_id)s
+         """
+    with cursor(database=database) as c:
+        c.execute(query, args={"bundle_id": bundle_id})
+        covariate_rows = list(c)
+        CODELOG.debug(
+            f"Downloaded {len(covariate_rows)} lines of study covariates for bundle_id {bundle_id} from '{database}'"
+        )
+
+    return covariate_rows
+
+
+def _upload_bundle_data_to_tier_3(cursor, model_version_id, bundle_data):
+    """Updloads bundle data to tier 3 attached to the current model_version_id.
+    """
+
+    insert_query = f"""
+    INSERT INTO epi.t3_model_version_dismod (
+        model_version_id,
+        bundle_id ,
+        seq ,
+        request_id ,
+        input_type_id ,
+        nid ,
+        underlying_nid ,
+        location_id,
+        sex_id ,
+        year_start ,
+        year_end ,
+        age_start,
+        age_end ,
+        measure_id ,
+        source_type_id ,
+        sampling_type_id,
+        representative_id ,
+        urbanicity_type_id ,
+        recall_type_id ,
+        recall_type_value ,
+        unit_type_id ,
+        unit_value_as_published ,
+        uncertainty_type_id ,
+        uncertainty_type_value ,
+        mean ,
+        lower ,
+        upper ,
+        standard_error ,
+        effective_sample_size ,
+        sample_size ,
+        cases ,
+        design_effect,
+        outlier_type_id
+    ) VALUES (
+        {model_version_id}, {", ".join(["%s"]*32)}
+    )
+    """
+
+    cursor.executemany(insert_query, bundle_data)
+
+    CODELOG.debug(f"uploaded {len(bundle_data)} lines of bundle data")
+
+
+def _upload_study_covariates_to_tier_3(cursor, model_version_id, covariate_data):
+    """Updloads study covariate mappings to tier 3 attached to the current model_version_id.
```

Comment:
Updloads -> Uploads

Reply:
I think that's actually an independent typo from the one above. I don't believe I copied and pasted this.